website/src/libs/api-v3/customExpressValidators.js

```diff
@@ -0,0 +1,46 @@
+import Q from 'q';
+import { model as User } from '../../models/user';
+
+export function isEqual (first, second) {
+  return first === second;
+}
+
+export function isUniqueUsername (username) {
+  if (!username) return true;
+
+  let lowerCaseUsername = username.toLowerCase();
+
+  let userQuery = User.findOne({
+    'auth.local.lowerCaseUsername': lowerCaseUsername,
+  }).exec();
+
+  return Q.promise((resolve, reject) => {
+    userQuery.then((user) => {
+      if (user) {
+        reject();
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+export function isUniqueEmail (email) {
+  if (!email) return true;
```

Comment:
We have to return true here, otherwise it tries to call `toLowerCase()` on an undefined variable. If we return true, than this validation passes and is not added as a validation error. If we return false, than the "email is taken" message appears for not providing an email, which seemed wrong to me.